pkg/model/fileshareproto/model.proto

```diff
@@ -0,0 +1,114 @@
+// This source file has been modified by 2019 OpenSDS Authors.
+// Copyright (c) 2019 OpenSDS Authors. All Rights Reserved.
+//
+// Copyright 2015, Google Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+syntax = "proto3";
+
+package fileshareproto;
+
+
+service Controller {
+	
+    // Create a file share
+    rpc CreateFileShare (CreateFileShareOpts) returns (GenericResponse){}
+ 
+}
+
+service FProvisionDock {
+    
+    // Create a file share
+    rpc CreateFileShare (CreateFileShareOpts) returns (GenericResponse){}
+
+}
+
+// CreateFileShareOpts is a structure which indicates all required properties for creating a file share.
+message CreateFileShareOpts {
+    // The uuid of the file share, optional when creating.
+    string id = 1;
+    // The name of the file share, required.
+    string name = 2;
+    // The requested capacity of the file share, required.
+    int64 size = 3;
+    // The description of the file share, optional.
```

Comment:
`CreateFileShareOpts ` lacks `shareProto ` filed which represents `CIFS `or `NFS `file share protocol. `Oceanstor `file driver need this filed. If there is a multi-tenant scenario, please add `tenantID `or `userID `to `CreateFileShareOpts ` .

Reply:
I am closing this PR as it has merge issues and invalid branch latest code change all other details,